crates/types/src/config/common.rs

```diff
@@ -420,6 +421,13 @@ pub enum LogFormat {
     Json,
 }
 
+pub trait CommonClientConnectionOptions {
```

Comment:
Overkill perhaps? It's nice to be able to use `create_tonic_channel_from_advertised_address` with either networking or metadata client options without additional conversions.

Reply:
Maybe we move this trait to where we define `create_tonic_channel_from_advertised_address` because it seems to belong to this method. Then we can also move the implementations there to keep things cohesive.

Reply:
I guess it did not work to create a `ConnectionOptions` struct which could be shared between the `MetadataStoreClientOptions` and the `NetworkingOptions`?

Reply:
Alrighty. Thanks for looking into it.